### ISSUE TITLE
Exception in thread "main" java.lang.ClassCastException

### ISSUE THREAD

**Issue by dancesWithCycles** [mecatran/gtfsvtor#69]
Hi folks,
Thank you so much for providing and maintaining this repo. I was calling gtfsvtor today getting this exception.

```
Exception in thread "main" java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
	at com.mecatran.gtfsvtor.validation.impl.ValidatorInjector.configureValidator(ValidatorInjector.java:201)
	at com.mecatran.gtfsvtor.validation.impl.ValidatorInjector.scanPackageAndInject(ValidatorInjector.java:110)
	at com.mecatran.gtfsvtor.validation.DefaultDaoValidator.<init>(DefaultDaoValidator.java:14)
	at com.mecatran.gtfsvtor.lib.GtfsVtor.validate(GtfsVtor.java:117)
	at comException in thread "main" java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
	at com.mecatran.gtfsvtor.validation.impl.ValidatorInjector.configureValidator(ValidatorInjector.java:201)
	at com.mecatran.gtfsvtor.validation.impl.ValidatorInjector.scanPackageAndInject(ValidatorInjector.java:110)
	at com.mecatran.gtfsvtor.validation.DefaultDaoValidator.<init>(DefaultDaoValidator.java:14)
	at com.mecatran.gtfsvtor.lib.GtfsVtor.validate(GtfsVtor.java:117)
	at com.mecatran.gtfsvtor.cmdline.GtfsVtorMain.main(GtfsVtorMain.java:56).mecatran.gtfsvtor.cmdline.GtfsVtorMain.main(GtfsVtorMain.java:56)
```

The instruction I called is the following.

```
~/gtfsvtor/gtfsvtor/bin/gtfsvtor -o vtor-res-top-level-dhid-nov-16-2021.html --numThreads 4  -v -c config.properties pxypihdrpv
```

I used the following config.

```
validator.CalendarValidator.checkEmptyCalendars=true
validator.CalendarValidator.checkExpired=true
validator.CalendarValidator.expiredCutoffDate=2020/10/30
validator.CalendarValidator.checkFuture=true
validator.CalendarValidator.futureCutoffDate=2022-10-29
validator.CalendarValidator.maxDaysWithoutService=7
validator.CalendarValidator.minDaysForCheckingNoServiceException=182
```

I am running this Java version.

```
java -version
openjdk version "11.0.12" 2021-07-20
OpenJDK Runtime Environment (build 11.0.12+7-post-Debian-2deb10u1)
OpenJDK 64-Bit Server VM (build 11.0.12+7-post-Debian-2deb10u1, mixed mode, sharing)
```

I could provide the GTFS file as well, but it has an inconvenient size of 65 mbytes.

Do you guys know anything that I possibly did wrong with my instruction and configuration?

I thank you so much for cooperation.

Cheers!

**Comment by dancesWithCycles** [comment]
Btw,
Here is the rest of the gtfsvtor instruction response.

```
GTFSVTOR version 0.1.1
Copyright (c) 2021 Mecatran
This program comes with absolutely no warranty.
This is free software, and you are welcome to redistribute it
under certain conditions; see the license file for details.

Loading config from config.properties
Loaded  agency.txt: 122 rows.
Loaded  routes.txt: 2990 rows.
Loaded  stops.txt: 33434 rows.
Loaded  calendar.txt: 2847 rows.
Loaded  calendar_dates.txt: 89794 rows.
Loaded  shapes.txt: 5958370 rows.
---[ Packing sorted shape points crude memory stats ]---
       What       |    Count   | Total (kB) | Per item  
------------------+------------+------------+-----------
     Shape points |    5958370 |      28242 |       4.85
           Shapes |      13717 |        107 |    2108.36
------------------+------------+------------+-----------
Loaded  trips.txt: 187062 rows.
Loaded  stop_times.txt: 4192794 rows.
Loaded  frequencies.txt: 0 rows.
Loaded  transfers.txt: 30091 rows.
------[ Packing sorted stop times crude memory stats ]-----
       What          |    Count   | Total (kB) | Per item  
---------------------+------------+------------+-----------
          Stop times |    4192794 |       6367 |       1.56
               Trips |     187062 |       4384 |      24.00
       Time patterns |      31155 |        629 |      20.68
       Stop patterns |      28029 |       1353 |      49.46
---------------------+------------+------------+-----------
Loaded 'pxypihdrpv' in 34683ms. Used memory: ~131Mb
```

**Comment by dancesWithCycles** [comment]
BTW, I get the same exception with Java 8.

```
~/gtfsvtor/gtfsvtor/bin/gtfsvtor -o vtor-res-top-level-dhid-nov-16-2021.html --numThreads 2  -v -c config.properties pxypihdrpv 
GTFSVTOR version 0.1.1
Copyright (c) 2021 Mecatran
This program comes with absolutely no warranty.
This is free software, and you are welcome to redistribute it
under certain conditions; see the license file for details.

Loading config from config.properties
Loaded  agency.txt: 122 rows.
Loaded  routes.txt: 2990 rows.
Loaded  stops.txt: 33434 rows.
Loaded  calendar.txt: 2847 rows.
Loaded  calendar_dates.txt: 89794 rows.
Loaded  shapes.txt: 5958370 rows.
---[ Packing sorted shape points crude memory stats ]---
       What       |    Count   | Total (kB) | Per item  
------------------+------------+------------+-----------
     Shape points |    5958370 |      28242 |       4.85
           Shapes |      13717 |        107 |    2108.36
------------------+------------+------------+-----------
Loaded  trips.txt: 187062 rows.
Loaded  stop_times.txt: 4192794 rows.
Loaded  frequencies.txt: 0 rows.
Loaded  transfers.txt: 30091 rows.
------[ Packing sorted stop times crude memory stats ]-----
       What          |    Count   | Total (kB) | Per item  
---------------------+------------+------------+-----------
          Stop times |    4192794 |       6367 |       1.56
               Trips |     187062 |       4384 |      24.00
       Time patterns |      31155 |        629 |      20.68
       Stop patterns |      28029 |       1353 |      49.46
---------------------+------------+------------+-----------
Loaded 'pxypihdrpv' in 37249ms. Used memory: ~155Mb
Exception in thread "main" java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
	at com.mecatran.gtfsvtor.validation.impl.ValidatorInjector.configureValidator(ValidatorInjector.java:201)
	at com.mecatran.gtfsvtor.validation.impl.ValidatorInjector.scanPackageAndInject(ValidatorInjector.java:110)
	at com.mecatran.gtfsvtor.validation.DefaultDaoValidator.<init>(DefaultDaoValidator.java:14)
	at com.mecatran.gtfsvtor.lib.GtfsVtor.validate(GtfsVtor.java:117)
	at com.mecatran.gtfsvtor.cmdline.GtfsVtorMain.main(GtfsVtorMain.java:56)
~/gtfs/vbn/top-level-dhid$ java -version
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)
```

**Comment by dancesWithCycles** [comment]
BTW, I'll get the same exception when I checkout the repository and call the same instruction again with gtfsvtor version 0.1.2.

```
~/gtfsvtor/build-tpx200-nov-16-2021/gtfsvtor/bin/gtfsvtor -o vtor-res-top-level-dhid-nov-16-2021.html --numThreads 2  -v -c config.properties pxypihdrpv 
GTFSVTOR version 0.1.2
Copyright (c) 2021 Mecatran
This program comes with absolutely no warranty.
This is free software, and you are welcome to redistribute it
under certain conditions; see the license file for details.

Loading config from config.properties
Loaded  agency.txt: 122 rows.
Loaded  routes.txt: 2990 rows.
Loaded  stops.txt: 33434 rows.
Loaded  calendar.txt: 2847 rows.
Loaded  calendar_dates.txt: 89794 rows.
Loaded  shapes.txt: 5958370 rows.
---[ Packing sorted shape points crude memory stats ]---
       What       |    Count   | Total (kB) | Per item  
------------------+------------+------------+-----------
     Shape points |    5958370 |      28242 |       4.85
           Shapes |      13717 |        107 |    2108.36
------------------+------------+------------+-----------
Loaded  trips.txt: 187062 rows.
Loaded  stop_times.txt: 4192794 rows.
Loaded  frequencies.txt: 0 rows.
Loaded  transfers.txt: 30091 rows.
------[ Packing sorted stop times crude memory stats ]-----
       What          |    Count   | Total (kB) | Per item  
---------------------+------------+------------+-----------
          Stop times |    4192794 |       6367 |       1.56
               Trips |     187062 |       4384 |      24.00
       Time patterns |      31155 |        629 |      20.68
       Stop patterns |      28029 |       1353 |      49.46
---------------------+------------+------------+-----------
Loaded 'pxypihdrpv' in 37627ms. Used memory: ~156Mb
Exception in thread "main" java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
	at com.mecatran.gtfsvtor.validation.impl.ValidatorInjector.configureValidator(ValidatorInjector.java:208)
	at com.mecatran.gtfsvtor.validation.impl.ValidatorInjector.scanPackageAndInject(ValidatorInjector.java:113)
	at com.mecatran.gtfsvtor.validation.DefaultDaoValidator.<init>(DefaultDaoValidator.java:14)
	at com.mecatran.gtfsvtor.lib.GtfsVtor.validate(GtfsVtor.java:117)
	at com.mecatran.gtfsvtor.cmdline.GtfsVtorMain.main(GtfsVtorMain.java:56)
:~/gtfs/vbn/top-level-dhid$ java -version
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)
```

**Comment by dancesWithCycles** [comment]
BTW, Gtfsvtor version 0.1.2 ran through when I left out the following config file.

```
validator.CalendarValidator.checkEmptyCalendars=true
validator.CalendarValidator.checkExpired=true
validator.CalendarValidator.expiredCutoffDate=2020/10/30
validator.CalendarValidator.checkFuture=true
validator.CalendarValidator.futureCutoffDate=2022-10-29
validator.CalendarValidator.maxDaysWithoutService=7
validator.CalendarValidator.minDaysForCheckingNoServiceException=182
```

**Comment by laurentg** [comment]
Indeed it's a wrong direct Long to int unboxing which is forbidden in Java:

    value = Integer.valueOf((int) value);

Which should be written as:

    value = Integer.valueOf((int) (long) value);

I'll correct it and submit the code, it's really a simple change.

**Comment by laurentg** [comment]
@dancesWithCycles Can you check with the latest head? Should solve the issue.

**Comment by dancesWithCycles** [comment]
Hi @laurentg ,
Thank you so much for the prompt help. No exception anymore with fix #69.

Cheers mate!
